src/core/control/tools/StrokeHandler.cpp

```diff
@@ -256,7 +182,9 @@ void StrokeHandler::onButtonReleaseEvent(const PositionInputData& pos, double zo
         const Point& pt = pv.front();
         if (this->hasPressure) {
             // Pressure inference provides a pressure value to the last event. Most devices set this value to 0.
-            this->stroke->setLastPressure(std::max(pt.z, pos.pressure) * this->stroke->getWidth());
+            const double newPressure = std::max(pt.z, pos.pressure * this->stroke->getWidth());
+            this->stroke->setLastPressure(newPressure);
+            this->viewPool->dispatch(xoj::view::StrokeToolView::THICKEN_FIRST_POINT_REQUEST, newPressure);
         }
         stroke->addPoint(pt);
```

Comment:
The issue with "dots" getting wrong bounds and being only drawn after the 5 seconds interval has passed can be fixed by changing this line:
```suggestion
       drawSegmentTo(pt);
```

Reply:
I con't reproduce the bug you are reporting. Also, this fix doesn't quite make sense: either the stroke is pressure-sensitive, in which case 
```
    this->viewPool->dispatch(xoj::view::StrokeToolView::THICKEN_FIRST_POINT_REQUEST, newPressure);
```
repaints the dot, or the stroke is not pressure sensitive, in which case nothing needs to be repainted: the dot should have already been painted upon ButtonPress.

Reply:
Oh, I see! The issue lies in `Stroke::updateBounds()` which fails to properly compute the stroke's bounds when `Stroke::addPoint()` adds the first point. So the initial repainted Range, based on `Stroke::boundingRect()`, is not the right one.

Reply:
It was a little bit more sneaky than that. I was basically doing `points.emplace_back(points.front())`, which is UB: `emplace_back()` caused the vector to reallocate so the reference `points.front()` is dangling by the time it's used.
This is fixed by d94e19e

Reply:
Unfortunately now the "dots" are rendered with delay on MacOS (where it worked previously with no delay). Something must still be wrong.

Reply:
I must correct myself. There is no difference between MacOS and Linux. Just for filled highlighter "dots" there is no immediate rendering (the "dot" is only rendered after the max. 5 second period). For filled pen "dots" there is no issue. I don't see where a flagDirtyRegion is apparent.

Reply:
@hrdl-github The nightly build works fine on MacOS (at least in that aspect).

Reply:
Fixed with the last commit. The `singleDot==true` case was not handled specifically in `StrokeToolFilledHighlighterView::draw()`. Thanks a lot for the catch and the investigation!

Reply:
> I found another issue when testing [e28d71d](https://github.com/xournalpp/xournalpp/commit/e28d71ddd13e9b5e4be061e045f31b0acf814f28): when creating strokes with > 1 dots using the filled highlighter I get a dotted boundary and no filling. When repainted a few seconds later I get the expected result.

I just wanted to report the same thing.

Reply:
^^Yeah, I realized this right fter posting. The "fixup" commit should solve this.

Reply:
Except that in https://github.com/xournalpp/xournalpp/commit/d94e19ead6ca56f1b11ef21c5a1198ca4673cf8d this was still fine, only in https://github.com/xournalpp/xournalpp/commit/e28d71ddd13e9b5e4be061e045f31b0acf814f28 it is not